### PR TITLE
✨ Type Declaration for v2

### PR DESCRIPTION
**Pull request by RFS-ADRENO**
I have updated basic type for the new version!

### CODE REVIEW

File: @types/index.d.ts
Comment (ThatXliner) [inline @types/index.d.ts:63]: Partial?

File: @types/index.d.ts
Comment (ThatXliner) [inline @types/index.d.ts:62]: TypeScript has method overloading. This could be rewritten as
```suggestion
ask(message: string): Promise<string>;
ask(message: string, config: TAskConfig & {format: EResponseType.JSON}): Promise<IAskResponseJSON>;
```
(Although I'm not too experienced with TypeScript so please double-check my syntax)

**Review by ThatXliner — CHANGES_REQUESTED** [review]

File: @types/index.d.ts
Comment (RFS-ADRENO) [reply @types/index.d.ts:62]: I have already thought about this, I also managed to make it work but there is one problem, its the intellisense will only recognize one of the two formats. I might be missing something, will look after this later on...

File: @types/index.d.ts
Comment (RFS-ADRENO) [reply @types/index.d.ts:63]: yes, all the properties are optional, I tested and it worked without any error

**Comment by EvanZhouDev** [comment]
Thank you all! I will merge after I have approval from both of you that the code is all good to go.

**Comment by EvanZhouDev** [comment]
@ThatXliner?

**Comment by Aldhanekaa** [comment]
Any Updates?? Can't wait for v2 to be released!

**Comment by EvanZhouDev** [comment]
@RFS-ADRENO I added better cookie support in 5e7bea0, and it may change up types a little bit. If you have time, tweak the types a bit, and I'll merge!

**Comment by RFS-ADRENO** [comment]
> @RFS-ADRENO I added better cookie support in [5e7bea0](https://github.com/EvanZhouDev/bard-ai/commit/5e7bea0c7df5bc20fbeba43175db5773278a926a), and it may change up types a little bit. If you have time, tweak the types a bit, and I'll merge!

Alright, I will have a look at it ASAP

**Comment by EvanZhouDev** [comment]
Thank you so much! You're awesome.

**Comment by EvanZhouDev** [comment]
:shipit: LGTM

**Pull request merged**
